### PR TITLE
SS Port: NPCAI & NPCPool

### LINKED ISSUE #5891: More SNPC issues
When setting an SNPC as a job manually they are naked.

When they die during they this sometimes loose thier head and keep going.

this happend
[02:27:15] Runtime in code/modules/surgery/organs/organ_internal.dm,42: The brain attempted to insert into a head, but head wasn't an organ! Carpet (151,41,2) (/turf/simulated/floor/carpet)
   src: The brain (/obj/item/organ/internal/brain)
   src.loc: Unknown (151,41,2) (/mob/living/carbon/human/interactive)

[02:27:15] Runtime in code/game/dna/dna2_helpers.dm,255: Attempting to reset DNA from a missing head!
   src: /datum/dna (/datum/dna)
   usr: Keagan Baskett (twinmold) (/mob/living/carbon/human/human)

### PR DESCRIPTION
**Pull request by variableundefined**
Port over Cyberboss's work in tgstation/tgstation#25429 / Port NPCAI / Fixes a bug with Equipping SNPC.

Fixes #5891

This port over NPC Pool to the Subsystem. And NPCAI. In addition, NPCAI no longer processes SNPC either, they are now completely isolated in two separate controller.

As this is my first time dealing with a controller, it's likely I forgot something, or left redundant code inside. It would be helpful if anyone can point them out.

🆑:
tweak: NPCpool (SNPC) is now a subsystem.
tweak: NPCAI (For simple mob) is now a subsystem.
fix: Customising SNPC no longer results in screaming death and a naked person.
/🆑

### CODE REVIEW

**Comment by variableundefined** [comment]
Should be ready for review now.

File: code/controllers/subsystem/npcpool.dm [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	// 3. Process delegation: if a bot (or bots) has been delegated, assign them to the task.
+	// 4. Process coordination: if a bot(or bots) has been asked to coordinate, assign them to help.
+	// 5. Do all assignments: goes through the delegated/coordianted bots and assigns the right variables/tasks to them.
+
+	if (!resumed)
+		src.currentrun = processing.Copy()
+		stage = PROCESSING_NPCS
+	//cache for sanic speed (lists are references anyways)
+	var/list/currentrun = src.currentrun
+	var/list/canBeUsed = src.canBeUsed
```
Comment (Crazylemon64) [inline code/controllers/subsystem/npcpool.dm]: not quite sure what's going on here

Comment (Crazylemon64) [reply code/controllers/subsystem/npcpool.dm]: tbh it'd be a good deal better if these used different varnames, so the caching would be more apparent

**Comment by variableundefined** [comment]
@Crazylemon64 

Addressed

**Pull request merged**